app/screens/apps_form/apps_form_field.tsx

```diff
@@ -0,0 +1,188 @@
+    switch (subtype) {
+        case 'email':
+            return 'email-address';
+        case 'number':
+            return 'numeric';
+        case 'tel':
+            return 'phone-pad';
+        case 'url':
+            return 'url';
+        default:
+            return 'default';
+    }
+}
+
+function AppsFormField({
+    field,
+    name,
+    errorText,
+    value,
+    onChange,
+    performLookup,
+}: Props) {
+    const theme = useTheme();
+    const style = getStyleSheet(theme);
+
+    const testID = `AppFormElement.${name}`;
+    const placeholder = field.hint || '';
+    const displayName = field.modal_label || field.label || '';
+
+    const handleChange = useCallback((newValue: string | boolean | string[]) => {
+        onChange(name, newValue);
+    }, [name]);
+
+    const getDynamicOptions = useCallback(async (userInput = ''): Promise<DialogOption[]> => {
+        const options = await performLookup(field.name, userInput);
+        return options.map((option) => ({
+            text: option.label,
+            value: option.value,
+        }));
+    }, [performLookup, field]);
+
+    switch (field.type) {
+        case AppFieldTypes.TEXT: {
+            return (
+                <TextSetting
+                    label={displayName}
+                    maxLength={field.max_length || (field.subtype === 'textarea' ? TEXTAREA_DEFAULT_MAX_LENGTH : TEXT_DEFAULT_MAX_LENGTH)}
+                    value={value as string}
+                    placeholder={placeholder}
+                    helpText={field.description}
+                    errorText={errorText}
+                    onChange={handleChange}
+                    optional={!field.is_required}
+                    multiline={field.subtype === 'textarea'}
+                    keyboardType={selectKeyboardType(field.subtype)}
+                    secureTextEntry={field.subtype === 'password'}
+                    disabled={Boolean(field.readonly)}
+                    testID={testID}
+                />
+            );
+        }
+        case AppFieldTypes.USER:
+        case AppFieldTypes.CHANNEL:
+        case AppFieldTypes.STATIC_SELECT:
+        case AppFieldTypes.DYNAMIC_SELECT: {
+            let dataSource = '';
+            let options: DialogOption[] = [];
+
+            switch (field.type) {
+                case AppFieldTypes.USER:
+                    dataSource = ViewConstants.DATA_SOURCE_USERS;
+                    break;
+                case AppFieldTypes.CHANNEL:
+                    dataSource = ViewConstants.DATA_SOURCE_CHANNELS;
+                    break;
+                case AppFieldTypes.DYNAMIC_SELECT:
+                    dataSource = ViewConstants.DATA_SOURCE_DYNAMIC;
+                    break;
+                case AppFieldTypes.STATIC_SELECT:
+                    if (field.options) {
+                        options = field.options.map((option) => ({text: option.label, value: option.value}));
+                    }
+            }
+
```

Comment:
I like how you pulled out `selectKeyboardType` and think somethink like that would be helpful here too so the `switch` statement is mostly returning a component.